core/src/lib.rs

```diff
@@ -75,6 +74,44 @@ pub use translation::address_translation;
 pub use transport::Transport;
 pub use upgrade::{InboundUpgrade, OutboundUpgrade, ProtocolName, UpgradeError, UpgradeInfo};
 
-#[derive(thiserror::Error, Debug)]
-#[error(transparent)]
-pub struct DecodeError(Box<dyn Error + Send + Sync + 'static>);
```

Comment:
I am inclined to leave this as, accept that we are removing the `UnwindSafe` auto trait and just do another major version bump. I don't think complicating the API with an additional enum is worth it at this stage.

@mxinden What do you think?

Reply:
Actually, lets do the following:

- Make the enum private and wrap it in a new type that doesn't expose anything.
- Record an issue to replace the enum with the inner error
- Track the issue in the `0.52` milestone
- Optional: Prepare a PR that fixes the above issue by removing the enum again and add that to `0.52` as well.

That way, we can merge this as a non-breaking change and merge in the breaking change as soon as we decide to do another breaking change. I think very likely, nobody actually depends on this error being `UnwindSafe`. I don't actually care about removing this auto trait but I do care about not making as many major version bumps.

Reply:
> Make the enum private and wrap it in a new type that doesn't expose anything.

Since the new type doesn't expose anything, I removed the enum.

I created https://github.com/libp2p/rust-libp2p/issues/3534 but can't add issues to milestones.

Reply:
Thank you, I've added it to the milestone!